src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/ReviewUnusedParameters.Fixer.cs

```diff
@@ -23,8 +31,144 @@ public sealed override FixAllProvider GetFixAllProvider()
         public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             // Fixer not yet implemented.
+            Diagnostic diagnostic = context.Diagnostics.Single();
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    MicrosoftMaintainabilityAnalyzersResources.ReviewUnusedParametersMessage,
+                    async ct => await RemoveNodes(context.Document, diagnostic, ct).ConfigureAwait(false), diagnostic.Id),
+                diagnostic);
+
             return Task.CompletedTask;
+        }
+
+        protected abstract SyntaxNode GetParameterNode(SyntaxNode node);
+
+        protected abstract bool CanContinuouslyLeadToObjectCreationOrInvocation(SyntaxNode node);
+
+        private async Task<Solution> RemoveNodes(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var solution = document.Project.Solution;
+            var pairs = await GetNodesToRemoveAsync(document, diagnostic, cancellationToken).ConfigureAwait(false);
+            foreach (var group in pairs.GroupBy(p => p.Key))
+            {
+                DocumentEditor editor = await DocumentEditor.CreateAsync(solution.GetDocument(group.Key), cancellationToken).ConfigureAwait(false);
+                // Start removing from bottom to top to keep spans of nodes that are removed later.
+                foreach (var value in group.OrderByDescending(v => v.Value.SpanStart))
+                {
+                    editor.RemoveNode(value.Value);
+                }
+
+                solution = solution.WithDocumentSyntaxRoot(group.Key, editor.GetChangedRoot());
+            }
+
+            return solution;
+        }
+
+        private ImmutableArray<IArgumentOperation>? GetOperationArguments(SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            while (node != null)
```

Comment:
It would be good to have an explanation of the intent of this method with a comment with suitable example at start of the method. You can probably mention different possibilities that the parameter `node` can be in relation to the parenting invocation and object creation.

Reply:
Thank you for adding such a detailed explanation!

---
In reply to: [161034919](https://github.com/dotnet/roslyn-analyzers/pull/1516#discussion_r161034919) [](ancestors = 161034919)